src/structures/GuildEmoji.js

```diff
@@ -110,10 +110,11 @@ class GuildEmoji extends Emoji {
    *   .catch(console.error);
    */
   edit(data, reason) {
+    data.roles = data.roles ? data.roles.map(r => r.id ? r.id : r) : undefined;
```

Comment:
1. This shouldn't mutate the arguments (If I happened to Object.freeze the data, this would error). Instead, declare it as a variable (my suggestion is `roles`).
1. You can optimize the calback doing `r => r.id || r`